src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs

```diff
@@ -11,71 +11,75 @@
+      AssertMetricDefinition(deserializedMetricDefinition, azureStorageQueueMetricDefinition);
+      var deserializedAzureStorageQueueMetricDefinition = deserializedMetricDefinition as StorageQueueMetricDefinition;
+      AssertAzureStorageQueueMetricDefinition(deserializedAzureStorageQueueMetricDefinition, azureStorageQueueMetricDefinition, deserializedMetricDefinition);
+    }
+
+    private static void AssertAzureStorageQueueMetricDefinition(StorageQueueMetricDefinition deserializedServiceBusMetricDefinition, StorageQueueMetricDefinition serviceBusMetricDefinition, MetricDefinition deserializedMetricDefinition)
+    {
+      Assert.NotNull(deserializedServiceBusMetricDefinition);
+      Assert.Equal(serviceBusMetricDefinition.AccountName, deserializedServiceBusMetricDefinition.AccountName);
+      Assert.Equal(serviceBusMetricDefinition.QueueName, deserializedServiceBusMetricDefinition.QueueName);
+      Assert.Equal(serviceBusMetricDefinition.SasToken, deserializedServiceBusMetricDefinition.SasToken);
+      Assert.NotNull(deserializedMetricDefinition.AzureMetricConfiguration);
+      Assert.Equal(serviceBusMetricDefinition.AzureMetricConfiguration.MetricName, deserializedMetricDefinition.AzureMetricConfiguration.MetricName);
+      Assert.NotNull(deserializedMetricDefinition.AzureMetricConfiguration.Aggregation);
+      Assert.Equal(serviceBusMetricDefinition.AzureMetricConfiguration.Aggregation.Type, deserializedMetricDefinition.AzureMetricConfiguration.Aggregation.Type);
+      Assert.Equal(serviceBusMetricDefinition.AzureMetricConfiguration.Aggregation.Interval, deserializedMetricDefinition.AzureMetricConfiguration.Aggregation.Interval);
+    }
+
+    private StorageQueueMetricDefinition GenerateBogusAzureStorageQueueMetricDefinition(string resourceGroupName)
+    {
+      var bogusAzureMetricConfiguration = GenerateBogusAzureMetricConfiguration();
+      var bogusGenerator = new Faker<StorageQueueMetricDefinition>()
+          .StrictMode(ensureRulesForAllProperties: true)
+          .RuleFor(metricDefinition => metricDefinition.Name, faker => faker.Name.FirstName())
+          .RuleFor(metricDefinition => metricDefinition.Description, faker => faker.Lorem.Sentence(wordCount: 6))
+          .RuleFor(metricDefinition => metricDefinition.ResourceType, faker => ResourceType.StorageQueue)
+          .RuleFor(metricDefinition => metricDefinition.AccountName, faker => faker.Name.LastName())
+          .RuleFor(metricDefinition => metricDefinition.QueueName, faker => faker.Name.FirstName())
+          .RuleFor(metricDefinition => metricDefinition.SasToken, faker => $"?sig={Base64UrlEncoder.Encode(faker.Lorem.Sentence(wordCount: 3))}")
+          .RuleFor(metricDefinition => metricDefinition.AzureMetricConfiguration, faker => bogusAzureMetricConfiguration)
+          .RuleFor(metricDefinition => metricDefinition.ResourceGroupName, faker => resourceGroupName)

```

Comment:
Please add this to our generic one as well

Reply:
There is not a test for `MetricsDeclarationWithGenericAzureYamlSerializationTests.cs` inside of `Tests/Serialization/MetricsDeclaration/`. Let's do a new issue for this

Reply:
:+1: